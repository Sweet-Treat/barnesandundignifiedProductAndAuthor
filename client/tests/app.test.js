import 'jsdom-global/register'; //at the top of file , even  , before importing react

import App from '../src/components/app.jsx';
import ProductDetails from '../src/components/productDetails.jsx';
import React from 'react';
import TestRenderer from 'react-test-renderer';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';


configure({ adapter: new Adapter() });

import { shallow, mount, render } from 'enzyme';

describe('App', () => {
  test('should render correctly in "debug" mode', async () => {
    const component = await shallow(<App debug />);

    expect(component).toMatchSnapshot();
  });
});

describe('App9', () => {

  test('snapshot renders', () => {
    const component = TestRenderer.create(<App />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the inner productDetails',  () => {
    //const wrapper = mount(<App />);
    const wrapper =  mount(<App debug />);

    expect(wrapper.find(ProductDetails).length).toEqual(1);

  });

  it('jhuguyguyguy guyguy', () => {
    const wrapper = mount(<App />);
    const bookWrapper = wrapper.find(ProductDetails);
    console.log(bookWrapper);

    //expect(bookWrapper.find({ 'pages': 435 })).to.have.lengthOf(1);
  });
});
