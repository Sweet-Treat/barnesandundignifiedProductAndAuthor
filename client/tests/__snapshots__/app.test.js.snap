// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;

exports[`App9 snapshot renders 1`] = `
<div>
  <h2
    className="text--center mb-s"
  >
    Product Details
  </h2>
  <table
    className="table"
  >
    <tbody>
      <tr>
        <th>
          ISBN-13: 
        </th>
        <td>
          
        </td>
      </tr>
      <tr>
        <th>
          Publisher: 
        </th>
        <td>
          <a
            href=""
          >
            
          </a>
        </td>
      </tr>
      <tr>
        <th>
          Publication Date: 
        </th>
        <td>
          Invalid Date
        </td>
      </tr>
      <tr>
        <th>
          Pages: 
        </th>
        <td />
      </tr>
      <tr>
        <th>
          Sales Rank: 
        </th>
        <td />
      </tr>
    </tbody>
  </table>
</div>
`;
